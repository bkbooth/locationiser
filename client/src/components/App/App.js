import React, { lazy, Suspense } from 'react';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import { ThemeProvider } from 'styled-components/macro';
import { theme } from 'utils/theme';
import Auth from 'components/Auth';
import Loader from 'components/Loader';
import Map from 'components/Map';
import GlobalStyle from 'components/styles/GlobalStyle';

const Home = lazy(() => import('pages/Home'));
const Login = lazy(() => import('pages/Login'));
const Signup = lazy(() => import('pages/Signup'));
const NotFound = lazy(() => import('pages/NotFound'));

function App() {
  return (
    <>
      <GlobalStyle />
      <ThemeProvider theme={theme}>
        <Router>
          <Map>
            <Auth>
              <Suspense fallback={<Loader />}>
                <Switch>
                  <Route exact path="/" component={Home} />
                  <Route exact path="/pin/:pinId" component={Home} />
                  <Route exact path="/login" component={Login} />
                  <Route exact path="/signup" component={Signup} />
                  <Route component={NotFound} />
                </Switch>
              </Suspense>
            </Auth>
          </Map>
        </Router>
      </ThemeProvider>
    </>
  );
}

export default App;
