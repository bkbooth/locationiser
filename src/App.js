import React from 'react';
import styled, { createGlobalStyle } from 'styled-components/macro';
import Auth from './components/Auth';
import Navbar from './components/Navbar';
import Home from './pages/Home';

const GlobalStyle = createGlobalStyle`
  html {
    box-sizing: border-box;
  }
  *, *:before, *:after {
    box-sizing: inherit;
  }

  body {
    font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen-Sans, Ubuntu, Cantarell, 'Helvetica Neue', sans-serif;
    margin: 0;
    padding: 0;
  }
`;

const Main = styled.div`
  padding: 20px 30px;
`;

function App() {
  return (
    <>
      <GlobalStyle />
      <Auth>
        <Navbar />
        <Main>
          <Home />
        </Main>
      </Auth>
    </>
  );
}

export default App;
