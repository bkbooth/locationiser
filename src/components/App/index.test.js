import React from 'react';
import { render } from '@testing-library/react';
import App from './index';

it('renders correctly', () => {
  const { container } = render(<App />);
  expect(container.cloneNode(true)).toMatchSnapshot();
});
