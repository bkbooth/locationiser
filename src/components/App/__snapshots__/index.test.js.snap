// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c15 {
  text-transform: uppercase;
  font-size: 0.7rem;
  -webkit-letter-spacing: 0.05rem;
  -moz-letter-spacing: 0.05rem;
  -ms-letter-spacing: 0.05rem;
  letter-spacing: 0.05rem;
  width: auto;
  background: #0094c0;
  color: #fbfbfb;
  padding: 10px 20px;
  border: none;
  border-radius: 5px;
  cursor: pointer;
  -webkit-transition: all 0.2s;
  transition: all 0.2s;
  margin-bottom: 10px;
}

.c15:last-child {
  margin-bottom: 0;
}

.c15:hover,
.c15:focus {
  box-shadow: 1px 1px 10px rgba(0,0,0,0.4);
  outline: none;
}

.c15:disabled {
  opacity: 0.7;
}

.c17 {
  text-transform: uppercase;
  font-size: 0.7rem;
  -webkit-letter-spacing: 0.05rem;
  -moz-letter-spacing: 0.05rem;
  -ms-letter-spacing: 0.05rem;
  letter-spacing: 0.05rem;
  width: auto;
  background: #0094c0;
  color: #fbfbfb;
  padding: 10px 20px;
  border: none;
  border-radius: 5px;
  cursor: pointer;
  -webkit-transition: all 0.2s;
  transition: all 0.2s;
  margin-bottom: 10px;
  background: #fbfbfb;
  color: #101010;
  border: 1px solid #656565;
}

.c17:last-child {
  margin-bottom: 0;
}

.c17:hover,
.c17:focus {
  box-shadow: 1px 1px 10px rgba(0,0,0,0.4);
  outline: none;
}

.c17:disabled {
  opacity: 0.7;
}

.c17:disabled {
  background: #ececec;
}

.c10 {
  display: inline-block;
  width: 110px;
  vertical-align: top;
  margin-top: 5px;
}

.c11 {
  font-family: -apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen-Sans,Ubuntu,Cantarell,'Helvetica Neue',sans-serif;
  font-size: 1rem;
  background: #fbfbfb;
  color: #101010;
  border: 1px solid #656565;
  border-radius: 5px;
  padding: 5px;
}

.c12 {
  font-family: -apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen-Sans,Ubuntu,Cantarell,'Helvetica Neue',sans-serif;
  font-size: 1rem;
  background: #fbfbfb;
  color: #101010;
  border: 1px solid #656565;
  border-radius: 5px;
  padding: 5px;
}

.c9 {
  margin-bottom: 10px;
}

.c7 {
  position: fixed;
  bottom: 0;
  width: 100vw;
  z-index: 3;
  background: #fbfbfb;
  padding: 20px;
  border-radius: 5px 5px 0 0;
  box-shadow: 0 10px 40px rgba(0,0,0,0.3);
  -webkit-transform: translateY(110%);
  -ms-transform: translateY(110%);
  transform: translateY(110%);
  -webkit-transition: -webkit-transform 0.5s;
  -webkit-transition: transform 0.5s;
  transition: transform 0.5s;
}

.c8 {
  font-size: 1.5rem;
  -webkit-letter-spacing: 0.1rem;
  -moz-letter-spacing: 0.1rem;
  -ms-letter-spacing: 0.1rem;
  letter-spacing: 0.1rem;
  text-transform: uppercase;
  color: #575757;
  margin-top: 0;
  margin-bottom: 30px;
  margin-bottom: 10px;
}

.c13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-top: 20px;
}

.c13 .c14,
.c13 .c16 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  margin-bottom: 0;
}

.c13 .c14:first-child,
.c13 .c16:first-child {
  margin-right: 10px;
}

.c0 {
  height: 100%;
  z-index: 1;
}

.c1 {
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: 2;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  pointer-events: none;
}

.c1 > * {
  pointer-events: auto;
}

.c2 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  width: 100%;
  max-height: calc(100% - 2 * 20px);
  overflow-y: auto;
  margin: 20px;
  border-radius: 5px;
  background: #fbfbfb;
  box-shadow: 0 10px 40px rgba(0,0,0,0.3);
}

.c6 {
  padding: 30px 20px;
}

.c6 *:first-child {
  margin-top: 0;
}

.c6 *:last-child {
  margin-bottom: 0;
}

.c3 {
  position: -webkit-sticky;
  position: sticky;
  top: 0;
  left: 0;
  right: 0;
  height: 70px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0 20px;
  background: #0094c0;
  color: #fbfbfb;
  box-shadow: 0 5px 10px rgba(0,0,0,0.4);
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0;
  font-size: 1.7rem;
  -webkit-letter-spacing: 0.05rem;
  -moz-letter-spacing: 0.05rem;
  -ms-letter-spacing: 0.05rem;
  letter-spacing: 0.05rem;
}

.c5 {
  width: 40px;
  height: auto;
  margin-right: 10px;
}

@media screen and (min-width:351px) {
  .c7 {
    left: calc(50% - 175px);
    width: 350px;
  }
}

@media screen and (min-width:540px) {
  .c2 {
    width: 500px;
    margin: 20px auto;
  }
}

<div>
  <div
    class="c0"
    id="leaflet-map"
  />
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <header
        class="c3"
      >
        <h1
          class="c4"
        >
          <img
            alt="world map emoji"
            class="c5"
            src="world-map-emoji.png"
          />
          locations
        </h1>
      </header>
      <div
        class="c6"
      >
        <p>
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-spinner-third fa-w-16 fa-spin "
            data-icon="spinner-third"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M456.433 371.72l-27.79-16.045c-7.192-4.152-10.052-13.136-6.487-20.636 25.82-54.328 23.566-118.602-6.768-171.03-30.265-52.529-84.802-86.621-144.76-91.424C262.35 71.922 256 64.953 256 56.649V24.56c0-9.31 7.916-16.609 17.204-15.96 81.795 5.717 156.412 51.902 197.611 123.408 41.301 71.385 43.99 159.096 8.042 232.792-4.082 8.369-14.361 11.575-22.424 6.92z"
              fill="currentColor"
            />
          </svg>
           Loading...
        </p>
      </div>
    </div>
  </div>
  <div
    class="c7"
  >
    <h1
      class="c8"
    >
      Create new pin
    </h1>
    <p>
      Drag 
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-map-marker-plus fa-w-12 "
        color="#38c000"
        data-icon="map-marker-plus"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 384 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M192 0C86.4 0 0 86.4 0 192c0 76.8 25.6 99.2 172.8 310.4 9.6 12.8 28.8 12.8 38.4 0C358.4 291.2 384 268.8 384 192 384 86.4 297.6 0 192 0zm112 200c0 8.84-7.16 16-16 16h-72v72c0 8.84-7.16 16-16 16h-16c-8.84 0-16-7.16-16-16v-72H96c-8.84 0-16-7.16-16-16v-16c0-8.84 7.16-16 16-16h72V96c0-8.84 7.16-16 16-16h16c8.84 0 16 7.16 16 16v72h72c8.84 0 16 7.16 16 16v16z"
          fill="currentColor"
        />
      </svg>
       marker on map to desired location
    </p>
    <form>
      <div
        class="c9"
      >
        <label
          class="c10"
          for="title"
        >
          Title
        </label>
        <input
          class="c11"
          id="title"
          name="title"
          type="text"
          value=""
        />
      </div>
      <div
        class="c9"
      >
        <label
          class="c10"
          for="description"
        >
          Description
        </label>
        <textarea
          class="c12"
          id="description"
          name="description"
          rows="3"
        />
      </div>
      <div
        class="c13"
      >
        <button
          class="c14 c15"
          disabled=""
          type="submit"
        >
          Create
        </button>
        <button
          class="c14 c16 c17"
        >
          Cancel
        </button>
      </div>
    </form>
  </div>
</div>
`;
